SpiceSharpTest/Components/Semiconductors/BJT/BJTTest.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.IO;
+using System.Text;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SpiceSharp;
+using SpiceSharp.Circuits;
+using SpiceSharp.Components;
+using SpiceSharp.IntegrationMethods;
+using SpiceSharp.Simulations;
+using SpiceSharpTest.Utils;
+
+namespace SpiceSharpTest.Components.Semiconductors.BJT
+{
+    [TestClass]
+    public class BJTTest
+    {
+        [TestMethod]
+        public void Emitter_Follower_DC()
+        {
+            // A test for BJT transistor (model mjd44h11) acting in an emitter-follower transistor circuit.
+            // A circuit consists of a BJT transistor and resistor connected to the transistor emitter.
+            // The voltage betwen base and emitter is around 0.7V when a transistor is in active mode.
+
+            var collectorVoltage = 100.0;
+            var baseVoltages = new double[] { 1.0, 2.0, 10.0, 20.00, 30.0, 60.0, 90.0, 98.0 };
+            var resistancesOfResistorConnectedToEmitter = new int[] { 5, 10, 30, 40, 50, 60, 70, 1000 };
+
+            foreach (var baseVoltage in baseVoltages)
+            {
+                foreach (var resistance in resistancesOfResistorConnectedToEmitter)
+                {
+                    double resistorCurrent = 0;
+
+                    var emitterVoltage =
+                        GetEmitterVoltageFromEmitterFollowerCircuit(
+                            baseVoltage,
+                            collectorVoltage,
+                            resistance,
+                            out resistorCurrent);
+
+                    var V_BE = baseVoltage - emitterVoltage;
+                    Assert.That.AreEqualWithTol(0.7, V_BE, 0, 0.5);
```

Comment:
I believe 0.5V is too high tolerance. I would like Spice# to be accurate (not an educational tool like the Falstad Java simulator), which means accuracy of the models is very important.
This is why I try to make the reference for tests as accurate as possible. In this case I'd use a simulation from ngSpice or Spice 3f5 with very strict tolerances as a reference (ref. #4 ).